test/functional/test_framework/test_node.py

```diff
@@ -140,6 +142,29 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             AddressKeyPair('mzRe8QZMfGi58KyWCse2exxEFry2sfF2Y7', 'cPiRWE8KMjTRxH1MWkPerhfoHFn5iHPWVK5aPqjW8NxmdwenFinJ'),
     ]
 
+    def is_valgrind_present(self):
+        """Check for valgrind presence"""
+        try:
+          version = subprocess.run(["valgrind", "--version"], stdout=subprocess.PIPE).stdout.decode("utf-8")
+          if version.find("valgrind", 0, 10) > -1:
+              return True
+          else:
+              return False
+        except OSError:
+          return False
+
+    def should_exit_valgrind_on_first_error(self):
+        """Determine if valgrind should run with --exit-on-first-error flag (>=3.15 runs with it while older versions don't)"""
+        version_regex = re.compile('[.bv]')
+        version_info = subprocess.run(["valgrind", "--version"], stdout=subprocess.PIPE).stdout.decode("utf-8").rstrip().split("valgrind-")[1:][0]
```

Comment:
I think this can be simplified. Only one execution of `valgrind --version` should be needed to a.) check if Valgrind is present and b.) to parse what version it is, so try to merge the logic of `is_valgrind_present` and `should_exit_valgrind_on_first_error`.

I suggest `get_valgrind_version` which could return the tuple `(valgrind_major, valgrind_minor)` if `valgrind` is installed or `None` otherwise.

Something like this:

```
valgrind_version = get_valgrind_version()
if valgrind_version and valgrind_version >= (3, 14):
    # add --exit-on-first-error=true
```

:)